std/regex/internal/backtracking.d

```diff
@@ -153,49 +156,64 @@ template BacktrackingMatcher(bool CTregex)
             memory = memory[2..$];
         }
 
-        void initialize(ref RegEx program, Stream stream, void[] memBlock)
+        void initialize(ref const RegEx program, Stream stream, void[] memBlock)
         {
-            re = program;
             s = stream;
             exhausted = false;
             initExternalMemory(memBlock);
             backrefed = null;
         }
 
-        auto dupTo(void[] memory)
+        override void dupTo(Matcher!Char m, void[] memBlock)
         {
-            typeof(this) tmp = this;
-            tmp.initExternalMemory(memory);
-            return tmp;
+            auto backtracking = cast(BacktrackingMatcher) m;
+            backtracking.s = s;
```

Comment:
`with (backtracking)` comes to mind

Reply:
but then
```d
with(backtracking) {
s = s;
}
```
???

Reply:
yah one name would need to be changed :)